docs/madr/decisions/014-circuit-breaker-and-outlier-detector-policies.md

```diff
@@ -0,0 +1,192 @@
+Consequences of this approach may introduce the situation when during reading envoy's stats for debugging purposes,
+you may look for stats prefixed with `cluster.<name>.circuit_breakers.` with occurred fault in mind, and according
+to the behaviour of our policy, if the data plane was ejected because of fault it's taking into account, stats which
+contain interesting us data will be prefixed with `cluster.<name>.outlier_detection.` prefix.
+
+## Considered Options
+
+* Create two separate policies: `CircuitBreaker` and `OutlierDetector`
+* Create single `CircuitBreaker` policy which would map 1:1 current policy
+* Create single `FaultDetector` policy which would also map 1:1 to the current `CircuitBreaker`, but with non misleading name
+
+## Decision Outcome
+
+Chosen option: create two separate -`CircuitBreaker` and `OutlierDetector` policies.
+
+## Positive Consequences
+
+* Having two separate policies, each with explicit configurations makes it easier to look for appropriate Envoy stats and XDP configuration values.
+* By combining two abstractions, which exist in Envoy, and calling them both combined with a name of one of these is confusing, especially when you are trying to debug some issue without deep understanding of internals of our service mesh.
+* Default `CircuitBreaker` policy is sufficient to replicate behaviour of current version of this policy - there won't be a need to include a default `OutlierDetector` policy
+
+## Negative Consequences
+
+* `CircuitBreaker` will be a policy with very small amount of configuration available, which may introduce some clutter (at some point there may be so many policies, that it will be much more difficult to understand all of them)
+
+## Solution
+
+### Current configuration
+
+Below is the sample `CircuitBreaker` configuration
+
+```yaml
+conf:
+  interval: 1s
+  baseEjectionTime: 30s
+  maxEjectionPercent: 20
+  splitExternalAndLocalErrors: false
+  thresholds:
+    maxConnections: 2
+    maxPendingRequests: 2
+    maxRequests: 2
+    maxRetries: 2
+  detectors:
+    totalErrors:
+      consecutive: 20
+    gatewayErrors:
+      consecutive: 10
+    localErrors:
+      consecutive: 7
+    standardDeviation:
+      requestVolume: 10
+      minimumHosts: 5
+      factor: 1.9
+    failure:
+      requestVolume: 10
+      minimumHosts: 5
+      threshold: 85
+
+```
+
+### Specification
+
+#### Top level
+
+Top-level targetRef can have all available kinds (with restriction of `MeshHTTPRoute`
+not being implemented as of time of writing this MADR):
+
+```yaml
+targetRef:
+  kind: Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute|MeshHTTPRoute
+  name: ...
+```
+
+#### From level
+
+`CircuitBreaker` and `OutlierDetector` are outbound only policies, so only `to` should be configured.
+
+#### To level
+
+```yaml
+to:
+- targetRef:
+    kind: Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute|MeshHTTPRoute
```

Comment:
Do we need `MeshGatewayRoute|MeshHTTPRoute`?

Reply:
I was thinking about it, but couldn't quickly figure a solid use case (especially that we don't currently have `MeshHTTPRoute`), so I decided to remove them from this MADR, and if there will be a valid use case, we can iterate.